src/Symfony/Component/ObjectMapper/ObjectMapper.php

```diff
@@ -92,21 +92,28 @@ public function map(object $source, object|string|null $target = null): object
 
         $this->objectMap[$source] = $mappedTarget;
         $ctorArguments = [];
-        $constructor = $targetRefl->getConstructor();
-        foreach ($constructor?->getParameters() ?? [] as $parameter) {
-            if (!$parameter->isPromoted()) {
-                continue;
-            }
-
+        $targetConstructor = $targetRefl->getConstructor();
+        foreach ($targetConstructor?->getParameters() ?? [] as $parameter) {
             $parameterName = $parameter->getName();
-            $property = $targetRefl->getProperty($parameterName);
 
-            if ($property->isReadOnly() && $property->isInitialized($mappedTarget)) {
-                continue;
+            if ($targetRefl->hasProperty($parameterName)) {
+                $property = $targetRefl->getProperty($parameterName);
+
+                if ($property->isReadOnly() && $property->isInitialized($mappedTarget)) {
+                    continue;
+                }
             }
 
-            // this may be filled later on see storeValue
-            $ctorArguments[$parameterName] = $parameter->isDefaultValueAvailable() ? $parameter->getDefaultValue() : null;
+            try {
+                $ctorArguments[$parameterName] = $this->getRawValue($source, $parameterName);
```

Comment:
Instead of using try/catch we could create a new method like this: 

```php
    private function isReadable(object $source, string $propertyName): mixed
    {
        if ($this->propertyAccessor) {
            return $this->propertyAccessor->isReadable($source, $propertyName);
        }

        if (!property_exists($source, $propertyName) && !isset($source->{$propertyName})) {
           return false;
        }

        return true;
    }
```

I'm not sure its better what do you think?

Reply:
Agree, fixed. It puts us more in control of being able to read a value.